src/main/java/de/hysky/skyblocker/mixins/SignEditScreenMixin.java

```diff
@@ -42,12 +59,31 @@ protected SignEditScreenMixin(Text title) {
 				}
 			}
 			//if the sign is being used to enter number send it to the sign calculator
-			if (messages[1].equals("^^^^^^^^^^^^^^^") && config.uiAndVisuals.inputCalculator.enabled) {
+			if (isInputSign() && config.uiAndVisuals.inputCalculator.enabled) {
 				SignCalculator.renderCalculator(context, messages[0], context.getScaledWindowWidth() / 2, 55);
 			}
 		}
     }
 
+	@Inject(method = "keyPressed", at = @At("HEAD"))
+	private void skyblocker$keyPressed(int keyCode, int scanCode, int modifiers, CallbackInfoReturnable<Boolean> cir) {
+		// enter key
+		if (keyCode != 257 || !Utils.isOnSkyblock() || !isInputSign()) {
+			return;
+		}
+
+		if (SkyblockerConfigManager.get().uiAndVisuals.inputCalculator.closeSignsWithEnter) {
+			ClientPlayNetworkHandler handler = MinecraftClient.getInstance().getNetworkHandler();
+			if (handler != null) {
+				// call close() first so that the calculation gets run
+				this.close();
+				handler.sendPacket(
+						new UpdateSignC2SPacket(this.blockEntity.getPos(), this.front, messages[0], messages[1], messages[2], messages[3])
+				);
```

Comment:
Please do not send a packet here. This is dangerous. `close` already sends a packet. No need for the network handler check either.